tests/benchmark/utils.py

```diff
@@ -150,41 +149,84 @@ def save_results_to_json(results, filename=None):
 
 def calculate_support_percentage(df):
     """Calculate the percentage of supported features (True) for each dtype in a DataFrame."""
-    feature_columns = df.drop(columns=['dtype'])
+    feature_columns = df.drop(columns=['dtype', 'sdtype'])
     # Calculate percentage of TRUE values for each row (dtype)
     percentage_support = feature_columns.mean(axis=1) * 100
-    return pd.DataFrame({'dtype': df['dtype'], 'percentage_supported': percentage_support})
+
+    df = pd.DataFrame({
+        'dtype': df['dtype'],
+        'sdtype': df['sdtype'],
+        'percentage_supported': percentage_support,
+    })
+    df['percentage_supported'] = df['percentage_supported'].round(2)
+    return df
+
+
+def _version_tuple(version_str):
+    return tuple(int(part) for part in version_str.split('.'))
 
 
 def compare_and_store_results_in_gdrive():
+    """Compare results, store them in Google Drive, and post a message to Slack.
+
+    This function compares the current results with previous ones and processes the
+    results to create a summary.
+    """
     csv_handler = CSVHandler()
     comparison_results = compare_previous_result_with_current()
 
     results = csv_handler.read('results/')
     sorted_results = {}
+    # Sort dictionary by version keys
+    results = {
+        key: value
+        for key, value in sorted(results.items(), key=lambda item: _version_tuple(item[0]))
+    }
+
+    # Compute the summary
+    summary = pd.DataFrame()
+    for name, df in results.items():
+        df = calculate_support_percentage(df)
+        if summary.empty:
+            summary = df.rename(columns={'percentage_supported': name})
+        else:
+            summary[name] = df['percentage_supported']
+
+    summary['average'] = summary[list(results)].mean(axis=1).round(2)
+    for col in summary.columns:
+        if col not in ('sdtype', 'dtype'):
+            summary[col] = summary[col].apply(lambda x: f'{x}%')
+
+    sorted_results['Summary'] = summary
 
     slack_messages = []
+    mark_results = {}
+    exit_code = 0
     for key, value in comparison_results.items():
         if not value.empty:
             sorted_results[key] = value
             if key == 'unsupported_dtypes':
                 slack_messages.append(':fire: New unsupported DTypes!')
+                mark_results['#EB9999'] = value
```

Comment:
minor: we should label the hex codes as constants like `RED_HEX = ...` or whatever it is